### PR TITLE
remove duplicate pageview counts

### PR DESCRIPTION
**Pull request by philschatz**
... by not tracking the canonical URL.

This is not an ideal solution because it causes us to track the non-canonical URL. However, there isn't really a good central place that fires off analytics tracking so this seemed like the minimal fix to address Marketing's concern of double-counting pageviews.

Refs #1601

# What it does with these changes

- `UA79034791` is the org-wide tracking code
- `UA302277982` is the book-specific tracking code

## When going to a page without any version numbers, the following are sent to `ga(...)`:

http://localhost:8000/contents/r-QzKsl_:JEgX-zGG

```
create UA-7903479-1 auto
UA79034791.send pageview /contents/r-QzKsl_:JEgX-zGG
create UA-30227798-2 auto
UA302277982.send pageview /contents/r-QzKsl_:JEgX-zGG
```

## When going to a page directly, the following are sent to `ga(...)`:

http://localhost:8000/contents/r-QzKsl_@7.23:JEgX-zGG@4/Introduction-to-Sociological-R

```
create UA-7903479-1 auto
UA79034791.send pageview /contents/r-QzKsl_@7.23:JEgX-zGG@4/Introduction-to-Sociological-R
create UA-30227798-2 auto
UA302277982.send pageview /contents/r-QzKsl_@7.23:JEgX-zGG@4/Introduction-to-Sociological-R
```

## When clicking `[Next]`, the following are sent to `ga(...)`:

```
create UA-7903479-1 auto
UA79034791.send pageview /contents/r-QzKsl_@7.23:JEgX-zGG/Introduction-to-Sociological-R
create UA-30227798-2 auto
UA302277982.send pageview /contents/r-QzKsl_@7.23:JEgX-zGG/Introduction-to-Sociological-R
```

## When going to just the book uuid URL:

http://localhost:8000/contents/afe4332a-c97f-4fc4-be27-4e4d384a32d8

```
create UA-7903479-1 auto
UA79034791.send pageview /contents/afe4332a-c97f-4fc4-be27-4e4d384a32d8
create UA-30227798-2 auto
UA302277982.send pageview /contents/afe4332a-c97f-4fc4-be27-4e4d384a32d8
```

## When clicking an item in the TableOfContents:

```
create UA-7903479-1 auto
UA79034791.send pageview /contents/r-QzKsl_@7.23:OkLwVQKH/Theoretical-Perspectives
create UA-30227798-2 auto
UA302277982.send pageview /contents/r-QzKsl_@7.23:OkLwVQKH/Theoretical-Perspectives
```

### CODE REVIEW

**Comment by reedstrm** [comment]
Hmm, seems to me we're sending "too early". Can we delay sending until we've rewritten the url to the final with versions form?

**Comment by tomekstach** [comment]
@philschatz please correct me, we will still have more then one address for the one page in the GA statistics? But we should not have many pageViews in the statistics which were caused by redirections?

If it's true, this solution might be ok for us. Of course, it would be ideal to have only one canonical address (in the GA statistics) for the one page.

**Comment by philschatz** [comment]
Analytics now sends the canonical URL. Analytics used to be sent whenever the router was triggered, now it is done explicitly after each page loads.

# Test Plan

This is for @kerwinso . Should it be on the Trello card rather than here?

## Ways to test

There are a couple of ways this could be tested:

1. uncomment the code in `analytics.coffee` which will print out what would have been sent to google analytics (but doesn't actually send)
1. always print what would be sent to google analytics **and** send it to google
1. click around and watch the **Realtime** view in Google Analytics for both cnx.org **and** the book you are viewing
1. Install a chrome extension like https://chrome.google.com/webstore/detail/google-analytics-debugger/jnkmfdileelhofjcijamephohjechhna that prints out what is being sent to google analytics

## Things to test

- every page type reports to the main cnx.org Google Analytics
  - `/`
  - `/search` add terms, click on links in the left nav bar, click the `[x]` to remove filters, click on a piece of content
  - `/contents`
  - `/browse` click on various subjects
  - `/tos`
  - `/license`
  - `/donate` and click the $10 Donate button
  - `/about` and click the Contact link
  - a 404 page
- Content
  - View a book that we own and verify both codes are sent the pageview
  - View a book that we do not own and verify both codes are sent the pageview
  - View a Page that is not in a book and verify the cnx code is sent
  - View a Page that is not in a book but has an analytics code associated with it and verify the code was sent
  - verify the canonical URL (book shortId + version and Page shortID + version) is the only one sent when visiting
    - a URL containing the full UUID of a book (no version)
    - a URL containing the shortId of a book (no version)
    - a URL containing the full UUID of a Page (no version)
    - a URL containing the shortId of a Page (no version)

**Comment by tomekstach** [comment]
@philschatz I will also try to test it tomorrow on a katalyst01. I am not sure if I will be able to test every single case - but I will try do test as much as I can.

File: src/scripts/loader.coffee [hunk 1]
```diff
@@ -91,7 +91,7 @@ define (require) ->
     if location.search
       router.navigate(Backbone.history.fragment, {replace: true})
 
-    analytics.sendAnalytics() # Track analytics for the initial page
+    # analytics.sendAnalytics() # Track analytics for the initial page
```
Comment (reedstrm) [inline src/scripts/loader.coffee]: commented out? Hmmm.

**Pull request merged**